### PR TITLE
Allow specific prune specific channel is a dict is defined

### PR DESCRIPTION
**Pull request by betoredhat**
The idea of this change is to start with the deprecation of a list of strings to define the operators to prune and start using a dictionary.

### CODE REVIEW

File: roles/prune_catalog/tasks/main.yml [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+        pc_operators | type_debug == "dict"
       - pc_source_catalog is defined
       - pc_destination_catalog is defined
       - pc_source_catalog | length > 0
       - pc_destination_catalog | length > 0
     fail_msg: "At least one of the parameters requirements is missing"
 
-- name: "Check if {{ pc_source_catalog }} is pullable"
-  shell: >
+- name: "Check if images is pullable - {{ pc_source_catalog }} "
```
Comment (ramperher) [inline roles/prune_catalog/tasks/main.yml:14]: Chec if images _are_ pullable

File: roles/prune_catalog/tasks/main.yml [hunk 2]
```diff
@@ -21,77 +22,82 @@
...
-- block:
-    - name: Create temporary directory
-      tempfile:
-        state: directory
-        prefix: pc_tmp_dir.
-      register: pc_tmp_dir
+- name: "Prune catalog"
+  when: skopeo_result.rc == 0
+  block:
+    # - name: Create temporary directory
```
Comment (ramperher) [inline roles/prune_catalog/tasks/main.yml:36]: Is this code (and the rest of commented code) commented for a specific reason? Can it be deleted?

File: roles/prune_catalog/tasks/main.yml (hunk 1 above)
Comment (betoredhat) [reply roles/prune_catalog/tasks/main.yml:14]: fixed

File: roles/prune_catalog/README.md [hunk 3]
```diff
@@ -44,6 +43,109 @@ The following application must be already present on the system.
...
+NAME                 DISPLAY NAME         DEFAULT CHANNEL
+compliance-operator  Compliance Operator  stable
+
+PACKAGE              CHANNEL  HEAD
+compliance-operator  stable   compliance-operator.v1.3.1
+```
+
+Prune a dictionary for specific channel
+```yaml
+- name: "Prune a all and/or single channels"
```
Comment (manurodriguez) [inline roles/prune_catalog/README.md:94]: Just wonder if this is a typo `Prune a all ...` maybe it's just `Prune all ...`

**Review by manurodriguez — APPROVED** [review]
LGTM, just minor comment

File: roles/prune_catalog/README.md (hunk 3 above)
Comment (betoredhat) [reply roles/prune_catalog/README.md]: yes, it was a typo. Thank!

File: roles/prune_catalog/README.md [hunk 4]
```diff
@@ -1,16 +1,14 @@
 # prune_catalog role
 
-A role to prune a File Base Catalogs (FBC), based on a list of required operators.
-
-By default, the catalog image sets a label "quay.expires-after" to define an expiration. The label can be removed or the time can be modified as needed.
+A role to prune a File Base Catalogs (FBC), based on a list of required operators. The role will prune all the channels if  `pc_operators` is a list or a specific channel if it is a dictionary.
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:3]:
```suggestion
A role to prune a File Base Catalog (FBC). The role creates a pruned catalog, leaving only the specified operators.
```

File: roles/prune_catalog/README.md [hunk 5]
```diff
@@ -1,16 +1,14 @@
...
 
 ## Parameters
 
 Name                   | Required | Default                                       | Description
 ---------------------- | -------- | --------------------------------------------- | ------------
 pc_source_catalog      | Yes      |                                               | Source catalog to be pruned
 pc_destination_catalog | Yes      |                                               | Catalog containing the required operators
-pc_operators_list      | Yes      |                                               | List of operators to include in the pruned catalog
+pc_operators           | Yes      |                                               | A List or dictionary of operators to include in the pruned catalog. See examples below
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:11]:
```suggestion
pc_operators           | Yes      |                                               | The set of operators to keep in the pruned catalog. See examples below
```

File: roles/prune_catalog/README.md [hunk 6]
```diff
@@ -44,6 +43,109 @@ The following application must be already present on the system.
...
+NAME                 DISPLAY NAME         DEFAULT CHANNEL
+compliance-operator  Compliance Operator  stable
+
+PACKAGE              CHANNEL      HEAD
+compliance-operator  4.7          compliance-operator.v0.1.32
+compliance-operator  release-0.1  compliance-operator.v0.1.61
+compliance-operator  stable       compliance-operator.v1.3.1
+```
+
+Prune a dictionary for specific channel
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:58]:
```suggestion
Create a pruned catalog with specific channels
```

File: roles/prune_catalog/README.md [hunk 7]
```diff
@@ -44,6 +43,109 @@ The following application must be already present on the system.
...
+```yaml
+- name: "Prune a catalog specific with channels"
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:60]:
```suggestion
- name: "Create a pruned catalog with specific channels"
```

File: roles/prune_catalog/README.md [hunk 8]
```diff
@@ -44,6 +43,109 @@ The following application must be already present on the system.
...
+      compliance-operator:
+        channel: stable
+      file-integrity-operator:
+        channel: stable
+      cluster-logging:
+        channel: stable
+    pc_opm_args: "--skip-tls-verify=true"
+```
+
+Only the specific channel for the listed operators are mirrored
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:76]:
```suggestion
Only the specified channel of the operators are kept in the catalog
```

File: roles/prune_catalog/README.md [hunk 9]
```diff
@@ -44,6 +43,109 @@ The following application must be already present on the system.
...
+
+$ DOCKER_CONFIG=/home/<user>/.docker oc-mirror list operators --catalog <my-local-registry>:4443/pruned-catalog:latest --package compliance-operator
+NAME                 DISPLAY NAME         DEFAULT CHANNEL
+compliance-operator  Compliance Operator  stable
+
+PACKAGE              CHANNEL  HEAD
+compliance-operator  stable   compliance-operator.v1.3.1
+```
+
+Prune a dictionary for specific channel
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:92]:
```suggestion
Another example of a pruned catalog with specific channels
```

File: roles/prune_catalog/README.md [hunk 10]
```diff
@@ -44,6 +43,109 @@ The following application must be already present on the system.
...
+```yaml
+- name: "Prune all and/or single channels"
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:94]:
```suggestion
- name: "Create a pruned catalog with custom channels"
```

File: roles/prune_catalog/README.md [hunk 11]
```diff
@@ -44,6 +43,109 @@ The following application must be already present on the system.
...
+NAME                     DISPLAY NAME             DEFAULT CHANNEL
+file-integrity-operator  File Integrity Operator  stable
+
+PACKAGE                  CHANNEL  HEAD
+file-integrity-operator  stable   file-integrity-operator.v1.3.3
+```
+
+### Catalog inspection
+
+Pruning an specific channels requires to be able to inspect the source index to identify the available channel to be pruned. oc-mirror can be used to perform the task. Eg:
```
Comment (tonyskapunk) [inline roles/prune_catalog/README.md:136]:
```suggestion
To keep specific channels, it is required to inspect the source catalog to identify the available channels. `oc-mirror` can be used to perform the task. Example:
```

**Review by tonyskapunk — APPROVED** [review]
LGTM

**Pull request merged**